addons/currency_rate_live/models/currency_rate_update.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+import json
+import urllib
+import xml.etree.ElementTree as ET
+
+from openerp.addons.web.controllers.main import xml2json_from_elementtree
+from openerp import api, fields, models
+
+
+class CurrencyRateProvider(models.Model):
+    _name = "currency.rate.provider"
+    _description = "Currency Rate Provider"
+
+    currency_ids = fields.Many2many('res.currency', string="Currencies to Update", required=True)
+    company_id = fields.Many2one('res.company', string="Company")
+    base_currency_id = fields.Many2one('res.currency', related='company_id.currency_id', store=True, string="Base Currency", required=True, readonly=True, default=1)
+    server_action_id = fields.Many2one('ir.actions.server', string='Name', domain=[('model_id', '=', 'currency.rate.provider')], required=True)
+
+    @api.multi
+    def process_currency_rate_data(self, request_url, response_type, preprocess):
```

Comment:
need to create separate method which fetch data from external resource with proper exception handling and then pass those data to process_currency_rate_data instead of pass request_url, response_type (do not process and fetch data in same method)